source/server-selection/server-selection.rst

```diff
@@ -1286,7 +1314,18 @@ Pseudocode for `single-threaded server selection`_::
 
             if servers is not empty:
                 in_window = servers within the latency window
-                return random entry from in_window
+                server1, server2 = random two entries from in_window
+
+                diff = abs(server1.pool.active_connection_count - server2.pool.active_connection_count)
+                if diff > max(0.05 * max_pool_size, 1):
+                    selected = server in (server1, server2) with minimum active_connection_count
+
+                if server1.pool.available_connection_count >= server2.pool.available_connection_count:
+                    selected = server1
+                else:
+                    selected = server2
+
+                return selected
```

Comment:
Can we define a shared method that implements `Selecting servers within the latency window` so that we don't need to duplicate the pseudocode here?

Reply:
done